src/optimize/bundles_route/dynamic_asset_response.js

```diff
@@ -0,0 +1,73 @@
+import { resolve } from 'path';
+import { open, fstat, createReadStream, close } from 'fs';
+
+import Boom from 'boom';
+import { fromNode as fcb } from 'bluebird';
+
+import { getFileHash } from './file_hash';
+import { replacePlaceholder } from '../public_path_placeholder';
+
+/**
+ *  Create a Hapi response for the requested path
```

Comment:
A bit more explanation maybe? Mention caching etc. Maybe pull in some of the info from your PR comment?